ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/AgentModelsSearcher.java

```diff
@@ -0,0 +1,37 @@
+package org.opensearch.ml.engine.tools;
+
+import static org.opensearch.ml.common.CommonValue.ML_AGENT_INDEX;
+import static org.opensearch.ml.common.CommonValue.TOOL_MODEL_RELATED_FIELD_PREFIX;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.ml.common.spi.tools.Tool;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+public class AgentModelsSearcher {
+    private Set<String> relatedModelIdSet;
+
+    public AgentModelsSearcher(Map<String, Tool.Factory> toolFactories) {
```

Comment:
Raw use of factory type is not the best practice. Can you redeclare the type as `Tool.Factory<? extends Tool>`?

Reply:
Oh, I found that this change impacts `MLAgentExecutor` and other downstream classees. I don't want to make the scope too large for this PR. Maybe you can leave it to other PRs.